### DISCUSSION TITLE
name in annotation shouldn't depends on following line.

### DISCUSSION THREAD

**Discussion by squirrelsc** [microsoft/nni#3765] [General]
In nni documents, it says name variable in nni.variable must be left variable of following line, and nni.functionchoice must be function name of right side. It doesn't make sense, and NNI can support annotation without name variable. The name variable should be used like display name in web ui only.

**Comment by Crysple** [comment]
This is just the guarantee that the user annotate on the right line(or on the right variable).

**Comment by squirrelsc** [comment]
> This is just the guarantee that the user annotate on the right line(or on the right variable).

user know where to add the line is enough. NNI shouldn't to make it harder to use to validate something right.
